### PR TITLE
Added customizable variable for shell command separator. Fixes #15

### LINKED ISSUE #15: Support fish shell
This works great!

However, it doesn't have support on the fish shell, because all the commands use `&&` that works in `bash` and `zsh`. 

Fish uses `; and` combination instead of `&&`

I replace all the `&&` combination like the following in `hexo.el`
```
hexo clean && hexo generate && hexo server --debug
```
to `; and` combination for this to work

```
hexo clean ; and hexo generate ; and hexo server --debug
```

I'm wondering if we can support  both `bash` and `fish` somehow

### PR DESCRIPTION
**Pull request by WarFox**
- Also added customizable variable for hexo server run and deploy commands
- hexo-shell-command-separator
- hexo--server-run-command
- hexo--server-deploy-command

### CODE REVIEW

File: hexo.el [hunk 1]
```diff
@@ -1153,21 +1153,36 @@ This is merely resonable for files in _posts/."
                             (hexo-find-command repo-path)
                             command-string))
 
+(defcustom hexo-shell-command-separator "&&"
+  "Default is &&, you may change it to `; and` in case you are using fish"
+  :group 'hexo
+  :type '(string))
+
+(defcustom hexo--server-run-command (format "hexo clean %s hexo generate %s hexo server --debug" hexo-shell-command-separator hexo-shell-command-separator)
```
Comment (kuanyui) [inline hexo.el:1161]: No good. You define this variable is a string list, but your default value is a string.

**Comment by kuanyui** [comment]
I will fix later. Close issue.

File: hexo.el (hunk 1 above)
Comment (WarFox) [reply hexo.el:1161]: `(format "hexo clean %s hexo generate %s hexo server --debug" hexo-shell-command-separator hexo-shell-command-separator)` is a string, not a string list

**Comment by WarFox** [comment]
okay, cool!

**Pull request closed without merging**